src/tests/banners/bannerSelection.tsx

```diff
@@ -32,37 +43,94 @@ const fetchBannerDeployTime = (region: ReaderRevenueRegion) => (): Promise<Date>
-    ),
-    'united-states': cacheAsync(
-        fetchBannerDeployTime('united-states'),
-        fiveMinutes,
-        'fetchBannerDeployTime_united-states',
-    ),
-    australia: cacheAsync(
-        fetchBannerDeployTime('australia'),
-        fiveMinutes,
-        'fetchBannerDeployTime_australia',
-    ),
-    'rest-of-world': cacheAsync(
-        fetchBannerDeployTime('rest-of-world'),
-        fiveMinutes,
-        'fetchBannerDeployTime_rest-of-world',
-    ),
+    contributions: {
+        'united-kingdom': cacheAsync(
+            fetchContributionsBannerDeployTime('united-kingdom'),
+            fiveMinutes,
+            'fetchContributionsBannerDeployTime_united-kingdom',
+        ),
+        'united-states': cacheAsync(
+            fetchContributionsBannerDeployTime('united-states'),
+            fiveMinutes,
+            'fetchContributionsBannerDeployTime_united-states',
+        ),
+        australia: cacheAsync(
+            fetchContributionsBannerDeployTime('australia'),
+            fiveMinutes,
+            'fetchContributionsBannerDeployTime_australia',
+        ),
+        'rest-of-world': cacheAsync(
+            fetchContributionsBannerDeployTime('rest-of-world'),
+            fiveMinutes,
+            'fetchContributionsBannerDeployTime_rest-of-world',
+        ),
+        // Contributions doesn't separate europe from row
+        'european-union': cacheAsync(
+            fetchContributionsBannerDeployTime('rest-of-world'),
+            fiveMinutes,
+            'fetchContributionsBannerDeployTime_rest-of-world',
+        ),
+    },
+    subscriptions: {
+        'united-kingdom': cacheAsync(
+            fetchSubscriptionsBannerDeployTime('united-kingdom'),
+            fiveMinutes,
+            'fetchSubscriptionsBannerDeployTime_united-kingdom',
+        ),
+        'united-states': cacheAsync(
+            fetchSubscriptionsBannerDeployTime('united-states'),
+            fiveMinutes,
+            'fetchSubscriptionsBannerDeployTime_united-states',
+        ),
+        australia: cacheAsync(
+            fetchSubscriptionsBannerDeployTime('australia'),
+            fiveMinutes,
+            'fetchSubscriptionsBannerDeployTime_australia',
+        ),
+        'rest-of-world': cacheAsync(
+            fetchSubscriptionsBannerDeployTime('rest-of-world'),
+            fiveMinutes,
+            'fetchSubscriptionsBannerDeployTime_rest-of-world',
+        ),
+        // Subscriptions separates europe from row
+        'european-union': cacheAsync(
+            fetchSubscriptionsBannerDeployTime('european-union'),
+            fiveMinutes,
+            'fetchSubscriptionsBannerDeployTime_european-union',
+        ),
+    },
 };
 
 // Has the banner been redeployed since the user last closed it?
-const redeployedSinceLastClosed = (targeting: BannerTargeting): Promise<boolean> => {
+export const redeployedSinceLastClosed = (
+    targeting: BannerTargeting,
+    bannerType: 'contributions' | 'subscriptions',
```

Comment:
Could introduce a named type for `'contributions' | 'subscriptions'` it is used in a few places.